service/topology_state_machine.hh

```diff
@@ -34,6 +34,7 @@ enum class node_state: uint16_t {
     replacing,           // the node replaces another dead node in the cluster and it data is being streamed to it
     rebuilding,          // the node is being rebuild and is streaming data from other replicas
     normal,              // the node does not do any streaming and serves the slice of the ring that belongs to it
+    left_token_ring,     // the node left the token ring, but not group0 yet; we wait until other nodes stop writing to it
```

Comment:
Does this break compatibility due to the assigned numbers changing?

Reply:
Everything in `topology_state_machine` is behind experimental flag and we're constantly doing incompatible changes in this area.

Reply:
This is bad practice. Sooner or later we'll do it to something that isn't, and that's a very difficult bug to recover from.

Reply:
I remember we did it for CDC, then group 0/schema-over-raft, now we're doing it for topology-over-raft. The code would quickly become completely unmaintainable if we had to introduce a cluster feature or something for every tiny modification we do. The point of experimental flag is to be able to develop a feature quickly, including breaking changes, before we consider it stable for release, no?

Reply:
> This is bad practice. Sooner or later we'll do it to something that isn't, and that's a very difficult bug to recover from.

The experimental mode exists specifically to not acquire technical dept we will not be able to get rid of afterwords. Some code is introduced with a knowledge that  incompatible modification will be done later.

Reply:
This isn't about a cluster feature, it's about not gratuitously changing an enum's meaning.

Reply:
The code is at a stage where we not only change enum's meaning, but also removing some values entirely. Even adding a new value without changing existing once requires cluster feature without experimental.